azuread/resource_application.go

```diff
@@ -213,3 +258,77 @@ func expandADApplicationHomepage(d *schema.ResourceData, name string) *string {
 
 	return p.String(fmt.Sprintf("https://%s", name))
 }
+
+func expandADApplicationRequiredResourceAccess(d *schema.ResourceData) *[]graphrbac.RequiredResourceAccess {
+	requiredResourcesAccesses := d.Get("required_resource_access").(*schema.Set).List()
+	result := make([]graphrbac.RequiredResourceAccess, 0)
+
+	for _, raw := range requiredResourcesAccesses {
+		requiredResourceAccess := raw.(map[string]interface{})
+		resource_app_id := requiredResourceAccess["resource_app_id"].(string)
+
+		result = append(result,
+			graphrbac.RequiredResourceAccess{
+				ResourceAppID: &resource_app_id,
+				ResourceAccess: expandADApplicationResourceAccess(
+					requiredResourceAccess["resource_access"].([]interface{}),
+				),
+			},
+		)
+	}
+	return &result
+}
+
+func expandADApplicationResourceAccess(in []interface{}) *[]graphrbac.ResourceAccess {
+	var resourceAccesses []graphrbac.ResourceAccess
+	for _, resource_access_raw := range in {
+		resource_access := resource_access_raw.(map[string]interface{})
+
+		resourceId := resource_access["id"].(string)
+		resourceType := resource_access["type"].(string)
+
+		resourceAccesses = append(resourceAccesses,
+			graphrbac.ResourceAccess{
+				ID:   &resourceId,
+				Type: &resourceType,
+			},
+		)
+	}
+
+	return &resourceAccesses
+}
+
+func flattenADApplicationRequiredResourceAccess(in *[]graphrbac.RequiredResourceAccess) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(*in))
+	if *in == nil {
+		return result
+	}
+
+	for _, requiredResourceAccess := range *in {
+		resource := make(map[string]interface{})
+		resource["resource_app_id"] = *requiredResourceAccess.ResourceAppID
```

Comment:
@tombuildsstuff I'm assuming here you're referring to the pointer to the field?  Just for my own knowledge, in the scenario the API returns an omitted ResourceAppID field, this would show up as a nil pointer and not an empty string?